securetar/__init__.py

```diff
@@ -180,25 +184,49 @@ class DecryptInnerTar:
 
             def __init__(self, parent: SecureTarFile) -> None:
                 """Initialize."""
+                self._head: bytes | None = None
                 self._parent = parent
                 self._pos = 0
                 self._size = tarinfo.size - IV_SIZE
                 self._tail = b""
 
             def read(self, size: int = 0) -> bytes:
                 """Read data."""
+                if self._head is None:
+                    # Read and validate header
+                    self._head = self._parent.read(max(size, 512))
+                    if (
+                        self._head[0 : len(GZIP_MAGIC_BYTES)] != GZIP_MAGIC_BYTES
+                        and self._head[
+                            TAR_MAGIC_OFFSET : TAR_MAGIC_OFFSET + len(TAR_MAGIC_BYTES)
+                        ]
+                        != TAR_MAGIC_BYTES
+                    ):
+                        raise ValueError("The inner tar is not gzip or tar, wrong key?")
```

Comment:
Maybe this should be extracted to a function?

Reply:
Yeah maybe? I mean technically someone could also put something else in such a tar, or use a different compression. We could say `decrypt()` is opinionated with what is typically used by this library, but having this separated seems like a good idea in any case.